Assets/Plugins/Source/Core/Utility/FileUtility.cs

```diff
@@ -55,6 +115,156 @@ public static string GenerateTempDirectory()
             return Path.GetFullPath(tempDirectory);
         }
 
+        /// <summary>
+        /// Get a list of the directories that are represented by the filepaths
+        /// provided. The list is unique, and is ordered by smallest path first,
+        /// so that the list can be utilized to create each directory in-order
+        /// if that is useful
+        /// </summary>
+        /// <param name="filepaths">
+        /// The filepaths to get the directories for.
+        /// </param>
+        /// <param name="inOrder">
+        /// If true, return the list of directories by order shortest path
+        /// first.
+        /// </param>
+        /// <returns></returns>
+        private static IEnumerable<string> GetDirectories(IEnumerable<string> filepaths, bool inOrder = true)
```

Comment:
I almost feel like `inOrder` needs renaming. It definitely does make things go in _an_ order, but isn't clear what kind. That's probably what your handy param documentation is for, though, so maybe not.

Reply:
I think a rename is appropriate. I think `creationOrder` would make sense, with notes indicating that it orders the directories in such a way that one can be created after the next. It just so _happens_ that order is also sorting the path strings by length in ascending order.